### PR TITLE
Change tab title after renaming discussion

### LINKED ISSUE #1521: After changing title, discussion title changes but page (tab) title stays the same
## Bug Report

**Current Behavior**
After a title change, the discussion title that is located under the tags does change, but the page (tab) title won't change until next refresh.

**Steps to Reproduce**
1. Go to a discussion
2. Click on edit title
3. Edit title

**Expected Behavior**
Page (tab) title should change with the discussion title.

**Screenshots**
htfdhfdhgdfhgfd

**Environment**
Flarum Community
https://discuss.flarum.org

### PR DESCRIPTION
**Pull request by datitisev**
**Fixes #1521**

**Changes proposed in this pull request:**
- Set title after renaming discussion

**Confirmed**

- [x] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `php vendor/bin/phpunit`).

### CODE REVIEW

**Comment by franzliedke** [comment]
What's with the lockfile change?

File: js/src/forum/components/RenameDiscussionModal.js [hunk 1]
```diff
@@ -58,6 +58,7 @@ export default class RenameDiscussionModal extends Modal {
           app.current.stream.update();
         }
         m.redraw();
+        app.setTitle(title());
```
Comment (franzliedke) [inline js/src/forum/components/RenameDiscussionModal.js]: Hmm, this feels wrong in terms of the virtual DOM / always re-render model. Shouldn't this title be set whenever the discussion page is re-rendered? (This is just one of the places where we trigger a redraw of that page - that redraw in turn should take care of the title change.)

**Comment by CDK2020** [comment]
@franzliedke It's from installing the npm modules, we can remove it if you need.

**Comment by datitisev** [comment]
@franzliedke Not sure. All it seems to do is modify a property for every package. All I did was run `npm install`, didn't update any package.

image

**Comment by franzliedke** [comment]
Strange. Probably a version conflict with NPM? Which version do you have?

**Comment by CDK2020** [comment]
@franzliedke it's version 6.2.0

File: js/src/forum/components/RenameDiscussionModal.js (hunk 1 above)
Comment (CDK2020) [reply js/src/forum/components/RenameDiscussionModal.js]: That's true, however if it did update this PR wouldn't exist

**Comment by datitisev** [comment]
@franzliedke I'll reset the lockfile changes if it's causing such a commotion 😂

File: js/src/forum/components/RenameDiscussionModal.js (hunk 1 above)
Comment (datitisev) [reply js/src/forum/components/RenameDiscussionModal.js]: Looks like the title is currently only set on https://github.com/flarum/core/blob/master/js/src/forum/components/DiscussionPage.js#L163 (DiscussionPage#show), which only runs when DiscussionPage#refresh runs, aka on first load of Discussion, I assume.

**Comment by tobscure** [comment]
Almost. Correct place to do this would be in `config` (theoretically `view` can be called without the view actually being rendered, whereas `config` is only called if the view has been rendered).

**Comment by datitisev** [comment]
@tobscure Done! Had to add a check for if `this.discussion` was set or not, seems like it was running before `DiscussionPage#show` was

**Pull request merged**